Initial version

doudou opened the pull request.

doudou committed:
Initial version

doudou committed:
add basic DS402 state machine support

doudou committed:
support Roboteq's mode-specific settings in the DS402 control word

doudou committed:
update for the final set/get/query API in canopen_master

g-arjones commented on src/Channel.hpp:
```diff
@@ -0,0 +1,102 @@
+    class Channel {
+        friend class Driver;
+
+        static const int ChannelObjectIDOffset = 0x800;
```
CHANNEL_OBJECT_ID_OFFSET

doudou committed:
update to inverted StateMachine/Driver dependency

canopen_master::Slave now gets its StateMachine object through
the constructor, and getStateMachine was removed. Update the code
